openhands/server/middleware.py

```diff
@@ -180,3 +181,23 @@ async def __call__(self, request: Request, call_next: Callable):
             await self._detach_session(request)
 
         return response
+
+
+class GitHubTokenMiddleware(SessionMiddlewareInterface):
+    def __init__(self, app):
+        self.app = app
+
+    async def __call__(self, request: Request, call_next: Callable):
+        settings_store = await shared.SettingsStoreImpl.get_instance(
+            shared.config, get_user_id(request)
+        )
+        settings = await settings_store.load()
+
+        # TODO: To avoid checks like this we should re-add the abilty to have completely different middleware in SAAS as in OSS
+        if not hasattr(request.state, "github_token"):
```

Comment:
```suggestion
        if not hasattr(request.state, "github_token") and not request.state.github_token:
```
We should also take into account the scenario when the state is actually present but `None`.

Reply:
Updated.